src/lib/ui/keyboard-avoiding-view/component.tsx

```diff
@@ -0,0 +1,47 @@
+import { KeyboardAwareScrollView } from '@pietile-native-kit/keyboard-aware-scrollview';
+import { ReactElement, ReactNode } from 'react';
+import { KeyboardAvoidingView, Platform, ScrollViewProps, StyleSheet, ViewStyle } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+export interface AppKeyboardAvoidingViewProps extends Partial<ScrollViewProps> {
```

Comment:
If `withoutScrollView` will be true, this component won't extend the `ScrollViewProps`. Could you please make an overload for props?

Example:

```tsx
export interface AppKeyboardAvoidingViewCommonProps {
  scrollEnabled?: boolean;
  extraHeight?: number;
  contentContainerStyle?: ViewStyle;
}

interface AppKeyboardAvoidingScrollableViewProps extends ScrollViewProps {
  scrollEnabled: true;
}

interface AppKeyboardAvoidingNonScrollableViewProps extends ViewProps {
  scrollEnabled?: boolean;
}

export type AppKeyboardAvoidingViewProps = AppKeyboardAvoidingViewCommonProps &
  (AppKeyboardAvoidingScrollableViewProps | AppKeyboardAvoidingNonScrollableViewProps);
```

Reply:
As discussed - removed